dataset_loaders/data_augmentation.py

```diff
@@ -562,6 +585,43 @@ def random_transform(x, y=None,
                                     fill_mode=fill_mode,
                                     fill_constant=cvalMask,
                                     rows_idx=rows_idx, cols_idx=cols_idx))
+    flow = None
+    if return_optical_flow:
+        from skimage import io
+        flow = []
+        if compute_optical_flow:
+            flow = optical_flow(x, rows_idx, cols_idx, chan_idx,
+                                return_rgb=return_optical_flow == 'rgb')
+        else:
+            if optical_flow_type not in ['Brox', 'Farn', 'LK', 'TVL1']:
+                raise RuntimeError('Unknown optical flow type')
+            optical_flow_path = os.path.join(data_path, 'OF',
+                                             optical_flow_type,
+                                             optical_flow_subdir)
+            if not os.path.exists(optical_flow_path):
+                raise RuntimeError('No optical flow found for this dataset')
+
+            for prefix, frame_names in sequence_names:
+                frame = prefix + '/' + frame_names
+                file_index = filenames.index(frame)
+                if file_index != 0:
+                    file_index -= 1
+                elif repeat_first_optical_flow:
+                    file_index = 0
+                else:
+                    of = np.zeros(x.shape[1:], x.dtype)
+                    flow.append(of)
+                    continue
+                frame = filenames[file_index]
+                of_path = os.path.join(optical_flow_path, frame + 'jpg')
+                if os.path.exists(of_path):
+                    of = io.imread(of_path)
+                    of = of.astype(x.dtype) / 255.
+                else:
+                    raise RuntimeError('Optical flow not found for this '
+                                       'file: %s' % of_path)
```

Comment:
I am having a really hard time understanding this code. can you please answer these questions when you have time?
1) What does `sequence_names` stand for?
2) This is my understanding of L606-622. Can you confirm it is correct and intended?
    ```python
    for _ in sequence_names:
        if file_index == 0 and not repeat_1st_opt_flow:
            of = np.zeros(x.shape[1:], x.dtype)
        else:
            if repeat_1st_opt_flow:
                frame = filenames[0]
            else:
                frame = filenames[-1]
            of_path = os.path.join(optical_flow_path, frame + 'jpg')
            of = io.imread(of_path).astype(x.dtype) / 255.
    ```
3) If what I wrote is correct, why do you set the first OF to zero when `repeat_1st_opt_flow` is False?
4) If what I wrote is correct, why do you return the OF of the last frame for every frame after the first, when `repeat_1st_opt_flow` is False?